### ISSUE TITLE
Port Master apps not working

### ISSUE THREAD

**Issue by thiago-123456** [cizia64/CrossMix-OS#8]
Port master seems to work, kind off...

It was able to open PM and it auto updated to 2024.05.31-1042 (beta channel selected as default in CrossMix).

All ports that I installed through PM failed to start (Super Mario War / Sonic Mania e etc.).

What am I doing wrong?

**Comment by cizia64** [comment]
Many PostMaster games are still not well configured for the TrimUI Smart Pro. So this question is probably more for the PortMaster team ;) 

I see that the Super Mario War script is not adapted to run on the TSP at all. It will not work as is.
Sonic Mania should run (I have added libcairo.so.2 lib which was missing to run it directly in CrossMix-OS). Note that Sonic Mania require that your copy your own assets.

Start simple and make a try with "Alchemy Quest"  from "ready to run" section, does it work ?

For information I've started this [compatibility list for PortMaster](https://docs.google.com/spreadsheets/d/1oCGa4lMDsZD4V7A559urMNcVzCZluptY8lyFmE45w5o/edit?usp=sharing)  help is welcome to complete the file and make some reports to the PostMaster team.

**Comment by cpal023** [comment]
I saw the compatibility list for PortMaster and a few of those games you have down as FALSE actually do run with a little tweaking. For example, Axiom Verge can run if you tweak the run script file. There is a command in there "|&" that is not recognized and you need to replace with a different value. Sonic Mania works if you delete the libtheora.dll as mention by @cobaltgit and others on that list work as well with some tinkering. I am sure in future updates these will be addressed. I cannot run Sonic 1,2, CD, Duke Nukem 3D AWO, Ships of Harkinian, or Streets of Rage Remake on the TSP. I have gotten them to work on Knulli but not stock or Tomato OS. About to this OS now. Would be cool to see a txt document of the required file extensions that this OS suports like Batocera has in each of the rom folders. For example MD-MSU and SNES-MSU file extensions. Just a suggestion. Exited to try this out!

**Comment by ogregorio** [comment]
If you were able to get them to work in Knuilli, I believe we could potentially port some libraries from there to improve compatibility here. I managed to run Sonic CD successfully, though I haven't gotten the sound to work yet.

**Comment by dxdatabase** [comment]
Same problem with the game To The Moon, bunch of missing sound libs. It run with Knulli after linking some libs, not with CrossMix. Libs missing :

- libsndfile
- libpulse
- libpulsecommon
- libpulse-simple
- libFLAC

**Comment by ogregorio** [comment]
Can you extract these libs from Knulli and package them for me? I can test whether it will work well in CrossMix.

**Comment by dxdatabase** [comment]
It don't work with CrossMix, not the same libc : 

/mnt/SDCARD/Data/ports/to_the_moon/gamedata/falcon_mkxp.bin: /lib64/libc.so.6: version `GLIBC_2.38' not found (required by /mnt/SDCARD/Data/ports/to_the_moon/libs/libpulsecommon-17.0.so)

If I check the glibc used by the Timui firmware : 
# /lib64/libc.so.6
GNU C Library (GNU) release release version 2.33.

And yes, more than less is better for the libs packed with Portmaster... But I don't have a suitable build environment to compile those libs.

**Comment by ogregorio** [comment]
You can use your TSP for this and enable opkg to download packages and libraries. However, I think this libc version can be ported for specific games by adding it inside the lib folder. Can you try this?

**Comment by ogregorio** [comment]
To update your OPKG repository list, follow these steps:

1. Edit the `distfeeds.conf` file located in `/ROOT/etc/opkg/`.
   
2. Replace the content with the following lines (using FTP is the simplest method):

   ```plaintext
   arch all 1
   arch noarch 1
   arch sunxi 10
   arch aarch64_cortex-a53 20

   src/gz reboot_core http://downloads.lede-project.org/snapshots/targets/sunxi/cortexa53/packages/
   src/gz reboot_base http://downloads.lede-project.org/snapshots/packages/aarch64_cortex-a53/base
   src/gz reboot_telephony http://downloads.lede-project.org/snapshots/packages/aarch64_cortex-a53/telephony
   src/gz reboot_packages http://downloads.lede-project.org/snapshots/packages/aarch64_cortex-a53/packages
   src/gz reboot_routing http://downloads.lede-project.org/snapshots/packages/aarch64_cortex-a53/routing
   src/gz reboot_luci http://downloads.lede-project.org/snapshots/packages/aarch64_cortex-a53/luci
   ```

3. Using SSH or Telnet, run the following commands:

   ```sh
   opkg update
   opkg install package1 package2
   ```

4. Optionally, you can also install additional packages like `make`:

   ```sh
   opkg install make
   ```
   
![image](https://github.com/cizia64/CrossMix-OS/assets/15088513/0eed1e47-d7c2-4246-8038-5638579d87e7)

**Comment by dxdatabase** [comment]
I've setup opkg, thanks. But there is something strange :


```
root@TinaLinux:~# opkg install nano
Installing nano (8.0-r1) to root...
Downloading http://downloads.lede-project.org/snapshots/packages/aarch64_cortex-a53/packages/nano_8.0-r1_aarch64_cortex-a53.ipk.
Installing libncurses6 (6.4-r2) to root...
Downloading http://downloads.lede-project.org/snapshots/packages/aarch64_cortex-a53/base/libncurses6_6.4-r2_aarch64_cortex-a53.ipk.
Configuring libncurses6.
Configuring nano.

root@TinaLinux:~# opkg files nano
Package nano (8.0-r1) is installed on root and has the following files:
/usr/bin/nano
root@TinaLinux:~# ls -l /usr/bin/nano
-rwxr-xr-x    1 root     root        196717 May  3 15:24 /usr/bin/nano
root@TinaLinux:~# /usr/bin/nano 
-ash: /usr/bin/nano: not found
```
Does internal storage have some kind of protection ?

**Comment by ogregorio** [comment]
If I am not mistaken, it is necessary to add /usr/bin to the PATH system.

**Comment by cizia64** [comment]
Yes may be some parts of the firmware are inn read only. These investigations are really interesting.

Now I have to learn to use opkg on the TSP 😄  (I've done that lopng time ago on some routers).
Some tips for missing libs : 


Use [batocera 32 64bits old image for Rpi 3](https://mirrors.o2switch.fr/batocera/bcm2837/stable/32/batocera-rpi3-32-20210926.img.gz) 
[pkgs.org](https://centos.pkgs.org/7/centos-aarch64/ImageMagick-6.9.10.68-3.el7.aarch64.rpm.html) propose a lot of binaries and libs. Choose oldest distrib to avoid problems with libc
[archlinuxarm.org](https://archlinuxarm.org/packages ) also really helpful with the same libc as our TSP.

If you have more similar tips do not hesitate.

The PortMaster launch script should be fixed for the TSP in the upstream !

**Comment by dxdatabase** [comment]
I did it ! Found libs for EasyRPG (framework used for To The Moon) on this repository : https://github.com/dokoma/kiwi/tree/main/Emus

![IMG_20240703_080627987](https://github.com/cizia64/CrossMix-OS/assets/173232307/48aaa987-58b8-4952-acd8-96ea1e700686)

There is a lot of interesting libs fully compatible with Crossmix's Portmaster to include in this repo !

**Comment by dxdatabase** [comment]
> I see that the Super Mario War script is not adapted to run on the TSP at all. It will not work as is.

Yes, the Trimui OS use Ash, the Busybox's shell. The Super Mario War script ask a lot of input to the user with dialog command. I don't know how to get user input with another program. And it use array, not implemented in ash. 

But, in the Kiwi repository, a real bash is available. I tried, and ... it don't work, because dialog is not displayed on screen.

**Comment by dxdatabase** [comment]
New SD card, fresh install of 1.2.1. And To The Moon don't work again 😭 

There's a nice installer made with love (uh, LÖVE framework). Again one missing lib (libsndfile), bet even if I copy it in the libs folder, the game crash : 

```
RGSS version 1 (XP) 
Failed to symlink /mnt/SDCARD/Data/ports/to_the_moon/conf//pulse/d81f93a4b2999dd50f9b932e669b44d7-runtime to /tmp/pulse-PKdhtXMmr18n: Operation not permitted
ALC_SOFT_pause_device present 
GL Vendor    : Imagination Technologies 
GL Renderer  : PowerVR Rogue GE8300 
GL Version   : OpenGL ES 3.2 build 1.19@6345021 
GLSL Version : OpenGL ES GLSL ES 3.20 build 1.19@6345021 
Failed to load libfluidsynth.so.1. Midi playback is disabled. 
EventThread termination requested 
RGSS thread ack'd request after 0 ms 
Shutting down. 
```

It worked on 1.1 just by adding missing libs... (if I remember, libfluidsynth.so.1 isn't mandatory to launch the game)

Problem with pulse's symlink ?

**Comment by Pietro395** [comment]
Has anyone managed to run Celeste and Stardew Valley? I am trying a lot of gamedata.

But where is it possible to see the logs?

**Comment by jarrard** [comment]
Marathon 1 works but the opengl mode has this weird tearing on the walls at medium distance.  Not sure how to fix. 

Anyone try and notice that issue with OPENGL enabled. (runs and plays better in OGL)
